ecs-init/engine/dependencies.go

```diff
@@ -36,6 +36,8 @@ type dockerClient interface {
 	RemoveExistingAgentContainer() error
 	StartAgent() (int, error)
 	StopAgent() error
+	LoadEnvVariables() map[string]string
```

Comment:
Rather than exposing both locations, I think it would be cleaner to have the logic of the merge fully encapsulated or moved into the config package.